src/base/ArrayUtils.hh

```diff
@@ -29,12 +29,15 @@ inline CELER_FUNCTION T norm(const array<T, N>& vec);
 
 //---------------------------------------------------------------------------//
 // Divide the given vector by its Euclidian norm
-inline CELER_FUNCTION void normalize_direction(array<real_type, 3>* direction);
+inline CELER_FUNCTION void normalize_direction(Real3* direction);
```

Comment:
Since you have an explicit type here (`Real3`), why is this specifically a pointer instead of a reference?

Reply:
Here I'm using the google-y convention of choosing pointers to be more obvious than mutable references. The change here was just to simplify the type name since it already has a typedef.

If you find an inconsistency in the references-as-pointers anywhere, please let me know!